tensorflow/lite/micro/docs/memory_management.md

```diff
@@ -94,6 +94,11 @@ integers of the following format:
 | ... | ... |
```

Comment:
I think before there were support for multiple subgraphs the intention was something like this.

| Offset | Value |
|-|-|
| 0 | Offline allocation format version |
| 1 | Subgraph index to which this allocation applies |   <--- SUBGRAPH INDEX 0
| 2 | Number offsets following: n |                       <--- NUMBER OF OFFSETS FOR SUBGRAPH 0
| 3 | Byte offset of tensor #0 or -1 to allocate at runtime |
| 4 | Byte offset of tensor #1 or -1 to allocate at runtime |
| ... | ... |
| 3+(n-1) | Byte offset of tensor #(n-1) or -1 to allocate at runtime |

| Offset | Value |
|-|-|
| 0 | Offline allocation format version |
| 1 | Subgraph index to which this allocation applies |    <--- SUBGRAPH INDEX 1
| 2 | Number offsets following: n |                        <--- NUMBER OF OFFSETS FOR SUBGRAPH 1
| 3 | Byte offset of tensor #0 or -1 to allocate at runtime |
| 4 | Byte offset of tensor #1 or -1 to allocate at runtime |
| ... | ... |
| 3+(n-1) | Byte offset of tensor #(n-1) or -1 to allocate at runtime |

Etc adding more of these...

However now it seems this is the current behavior and I think the documentation should be updated to something like this:

| Offset | Value |
|-|-|
| 0 | Offline allocation format version |
| 1 | Number of subgraphs: n |
| 2 | Number tensors |
| 3 | Byte offset for subgraph 0 of tensor #0 or -1 to allocate at runtime |
| 4 | Byte offset for subgraph 0 of tensor #1 or -1 to allocate at runtime |
| ... | ... |
| 3 + (num_tensors_for_subgraph_0 - 1) | Byte offset for subgraph 0 of tensor #(num_tensors_for_subgraph_0 - 1) or -1 to allocate at runtime |
| ... | Byte offset for subgraph 1 of tensor #0 or -1 to allocate at runtime |
| ... | Byte offset for subgraph 1 of tensor #1 or -1 to allocate at runtime |
| ... | ... |
| ... | Byte offset for subgraph 1 of tensor #(num_tensors_for_subgraph_1 - 1) or -1 to allocate at run
| ... | ... |
| ... | Byte offset for subgraph (n - 1) of tensor #0 or -1 to allocate at runtime |
| ... | Byte offset for subgraph (n - 1) of tensor #1 or -1 to allocate at runtime |
| ... | ... |
| ... | Byte offset for subgraph (n - 1) of tensor #(num_tensors_for_subgraph_x - 1) or -1 to allocate at run

Reply:
OK thank you for the explanation. I've updated the documentation slightly. I didn't put the full table in as you put it here, I think that will only be confusing, especially for the most common case with only a single subgraph. Let me know if the current documentation is still missing something.